lib/_http_outgoing.js

```diff
@@ -299,19 +302,45 @@ OutgoingMessage.prototype._renderHeaders = function _renderHeaders() {
 };
 
 OutgoingMessage.prototype.cork = function() {
+  this[kCorked]++;
   if (this.socket) {
     this.socket.cork();
-  } else {
-    this[kCorked]++;
   }
 };
 
 OutgoingMessage.prototype.uncork = function() {
+  this[kCorked]--;
   if (this.socket) {
     this.socket.uncork();
-  } else if (this[kCorked]) {
-    this[kCorked]--;
   }
+
+  if (this[kCorked] || this[kChunkedBuffer].length === 0) {
+    return;
+  }
+
+  const len = this[kChunkedLength];
+  const buf = this[kChunkedBuffer];
+
+  assert(this.chunkedEncoding);
+
+  let callbacks;
+  this._send(NumberPrototypeToString(len, 16), 'latin1', null);
+  this._send(crlf_buf, null, null);
+  for (let n = 0; n < buf.length; n += 3) {
+    this._send(buf[n + 0], buf[n + 1], null);
+    if (buf[n + 2]) {
+      callbacks ??= [];
+      callbacks.push(buf[n + 2]);
```

Comment:
Should we use `push` primordial here? From what I remember, this can cause some performance issues.

It's probably worth at least a comment if we avoid it.

Reply:
This is not an error path. As far as I am aware, per current consensus primordials does not apply here.

Reply:
@benjamingr